### PR TITLE
feat(c++,pt-expt): add .pt2 (AOTInductor) C/C++ inference with DPA1/DPA2/DPA3 support 

### LINKED ISSUE #5227: C++ Inference Support (Export + AOTI)
Support model inference in C++, leveraging exported models and AOTI.

**Goals:**
- Ensure the exported models can be loaded and executed in C++.
- Validate performance and correctness against Python inference.\n\n---\n*Generated by OpenClaw*

### PR DESCRIPTION
**Pull request by wanghan-iapcm**
Add C/C++ inference support for the `.pt2` (torch.export / AOTInductor) backend, covering all major descriptor types: SE_E2_A, DPA1, DPA2, and DPA3.                                                                                                         
                                                                                                                                                                                                                                                             
  ### C/C++ inference backend (`DeepPotPTExpt`)                                                                                                                                                                                                                
  - New `DeepPotPTExpt` backend that loads `.pt2` models via `torch::inductor::AOTIModelContainerRunnerCpu`                                                                                                                                                  
  - Supports PBC, NoPbc, fparam/aparam, multi-frame batching, atomic energy/virial, LAMMPS neighbor list (with ghost atoms, 2rc padding, type selection)
  - Registered alongside existing PT/TF/JAX/PD backends via the `.pt2` file extension

  ### dpmodel fixes for torch.export compatibility
  - Replace `[:, :nloc]` slicing with `xp_take_first_n()` in DPA1, DPA2, DPA3, and repflows/repformers — the original slicing creates `Ne(nall, nloc)` shape constraints that fail when `nall == nloc` (NoPbc case)
  - Replace flat `(nf*nall,)` indexing in `dpa1.py` and `exclude_mask.py` with `xp_take_along_axis`
  - Replace `xp.reshape(mapping, (nframes, -1, 1))` with `xp.expand_dims` in repflows/repformers — the `-1` resolves to `nall` during tracing

  ### pt_expt serialization
  - `.pt2` export via `torch.export.export` → `aot_compile` → package as zip
  - Python inference via `torch._inductor.aoti_load_package`

  ### Bug fix in all C++ backends
  - Fix ghost-to-local mapping when virtual atoms are present — the old code `mapping[ii] = lmp_list.mapping[fwd_map[ii]]` used post-filter indices as original indices; fixed to `mapping[ii] = fwd_map[lmp_list.mapping[bkw_map[ii]]]`
  - Fix use-after-free in `DeepPotPTExpt.cc` where `torch::from_blob` referenced a local vector after it went out of scope

  ### Test infrastructure
  - Model generation scripts (`gen_dpa1.py`, `gen_dpa2.py`, `gen_dpa3.py`, `gen_fparam_aparam.py`) that build from dpmodel config → serialize → export to both `.pth` and `.pt2` with identical weights
  - Remove pre-committed `.pth` files; regenerate in CI via `convert-models.sh`
  - C++ tests for all descriptor types: SE_E2_A, DPA1, DPA2, DPA3 (both `.pth` and `.pt2`, PBC + NoPbc, double + float)
  - Python unit tests for pt_expt inference (`test_deep_eval.py`)


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added support for PyTorch exportable (.pt2) models and runtime detection, enabling AOTInductor-based inference across interfaces.

* **Bug Fixes**
  * Improved neighbor/embedding extraction and broadcasting to increase backend export compatibility and robustness.

* **Tests**
  * Added extensive C++ and Python test suites and reference-generation scripts to validate .pt2 inference paths and cross-format consistency.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

### CODE REVIEW

File: source/api_cc/src/DeepPotPTExpt.cc [hunk 1]
```diff
@@ -0,0 +1,1296 @@
...
+  std::string content((std::istreambuf_iterator<char>(ifs)),
+                      std::istreambuf_iterator<char>());
+  ifs.close();
+
+  // Simple ZIP central directory parser
+  // Find End of Central Directory Record (EOCD)
+  // EOCD signature: 0x06054b50
+  size_t eocd_pos = std::string::npos;
+  for (size_t i = content.size() - 22; i > 0 && i != std::string::npos; --i) {
+    if (content[i] == 0x50 && content[i + 1] == 0x4b &&
```
[the remark this answers is not in the document]
Comment (wanghan-iapcm) [reply source/api_cc/src/DeepPotPTExpt.cc]: fixed by a3ed6d69

File: source/api_cc/src/DeepPotPTExpt.cc [hunk 2]
```diff
@@ -0,0 +1,1296 @@
...
+        if (field_id == 0x0001) {  // ZIP64 extra field
+          size_t field_data = extra_pos + 4;
+          int offset_in_field = 0;
+          if (uncompressed_size == 0xFFFFFFFF) {
+            uncompressed_size = 0;
+            for (int b = 0; b < 4; ++b) {
+              uncompressed_size |=
+                  static_cast<uint32_t>(static_cast<unsigned char>(
+                      content[field_data + offset_in_field + b]))
+                  << (8 * b);
```
[the remark this answers is not in the document]
Comment (wanghan-iapcm) [reply source/api_cc/src/DeepPotPTExpt.cc]: fixed by  45b8fc2f

**Review by iProzd — APPROVED** [review]

**Review by njzjz — APPROVED** [review]
